### PR TITLE
Add CI for checking registered data_type of new Op

### PR DESCRIPTION
**Pull request by Aurelius84**
**This PR is mainly used to enhance checking registered data_type of new Op.**

Here are some cases that **would be not** allowed:
+ **Only one data_type is registered of new op in PR.**
```cpp
// one of float, double, int, int64_t
REGISTER_OP_CPU_KERNEL(
    expand_test_e, ops::ExpandKernel<paddle::platform::CPUDeviceContext, int>); 
REGISTER_OP_CPU_KERNEL(
    expand_test_e_grad,
    ops::ExpandGradKernel<paddle::platform::CPUDeviceContext, int>); 
```
+ **Only one of float/double(or int/int64_t)  is registered of new op in PR.**
```cpp
// If you register int, the int64_t should be registered too.
REGISTER_OP_CPU_KERNEL(
    expand_test_e, ops::ExpandKernel<paddle::platform::CPUDeviceContext, int>,
    ops::ExpandKernel<paddle::platform::CPUDeviceContext, float>); 
REGISTER_OP_CPU_KERNEL(
    expand_test_e_grad,
    ops::ExpandGradKernel<paddle::platform::CPUDeviceContext, int>,
    ops::ExpandGradKernel<paddle::platform::CPUDeviceContext, float>); 
```

However, the followed case is allowed:
```cpp
// It looks good, because all needed data_types are registered.
REGISTER_OP_CPU_KERNEL(
    expand_test_e, ops::ExpandKernel<paddle::platform::CPUDeviceContext, float>,
    ops::ExpandKernel<paddle::platform::CPUDeviceContext, double>,
    ops::ExpandKernel<paddle::platform::CPUDeviceContext, int>,
    ops::ExpandKernel<paddle::platform::CPUDeviceContext, int64_t>); 
REGISTER_OP_CPU_KERNEL(
    expand_test_e_grad,
    ops::ExpandGradKernel<paddle::platform::CPUDeviceContext, float>,
    ops::ExpandGradKernel<paddle::platform::CPUDeviceContext, double>,
    ops::ExpandGradKernel<paddle::platform::CPUDeviceContext, int>,
    ops::ExpandGradKernel<paddle::platform::CPUDeviceContext, int64_t>); 
```
or
```cpp
// It looks good, because both float and double are registered.
REGISTER_OP_CPU_KERNEL(
    expand_test_e, ops::ExpandKernel<paddle::platform::CPUDeviceContext, float>,
    ops::ExpandKernel<paddle::platform::CPUDeviceContext, double>); 
REGISTER_OP_CPU_KERNEL(
    expand_test_e_grad,
    ops::ExpandGradKernel<paddle::platform::CPUDeviceContext, float>,
    ops::ExpandGradKernel<paddle::platform::CPUDeviceContext, double>); 
```
or
```cpp
// It looks good, because both int and int64_t are registered.
REGISTER_OP_CPU_KERNEL(
    expand_test_e, ops::ExpandKernel<paddle::platform::CPUDeviceContext, int>,
    ops::ExpandKernel<paddle::platform::CPUDeviceContext, int64_t>); 
REGISTER_OP_CPU_KERNEL(
    expand_test_e_grad,
    ops::ExpandGradKernel<paddle::platform::CPUDeviceContext, int>,
    ops::ExpandGradKernel<paddle::platform::CPUDeviceContext, int64_t>); 
```

**For some special op, only one of `float/double` and `int/int64_t` is supported. You can request some specific reviewer to approve you PR.**

### Examples PR
+ please see [log](http://ci.paddlepaddle.org/viewLog.html?buildId=236558&tab=buildLog&buildTypeId=YYTest_PrCiCpuPy2&logTab=tree&filter=all)  of [PR 21526](https://github.com/PaddlePaddle/Paddle/pull/21526) in details.
In this PR, five new operators are added and their registered data_type as follows:
```cpp
expand_test_a -> float int   // not allowed
expand_test_b -> float double int  // not allowed
expand_test_c -> float  // not allowed
expand_test_d -> float double int int64_t 
expand_test_e -> int int64_t
```

In above ops, `expand_test_a`、`expand_test_b` and`expand_test_c` are illegal, so gives warnings in log and needs specific reviewer to approve.
**The warnings info: **
```
[Step 1/1] 0 . More data_type of new operator should be regitered in your PR. Please make sure that both float/double (or int/int64_t) have been regitered. You must have one RD (Aurelius84 or liym27 or zhhsplendid)approval for the data_type registration of new operator.
[03:56:30]	[Step 1/1] 
[03:56:30]	[Step 1/1] There are 1 approved errors.
[03:56:30]	[Step 1/1] ****************
[03:56:31]	[Step 1/1] + expand_test_a float int
[03:56:31]	[Step 1/1] + expand_test_b double float int
[03:56:31]	[Step 1/1] + expand_test_c float
[03:56:39]	[Step 1/1] Process exited with code 1
```

image

**If  these new ops only supporte one of `float/double` and `int/int64_t` . You can request some specific reviewer to approve you PR.**  In above [PR 21526](https://github.com/PaddlePaddle/Paddle/pull/21526), if one of `Aurelius84/lym27/zhhsplendid `approve it. The CI will pass and log looks like [this](http://ci.paddlepaddle.org/viewLog.html?buildId=236715&buildTypeId=YYTest_PrCiCpuPy2&tab=buildLog) in details.

### CODE REVIEW

**Review by luotao1 — APPROVED** [review]
LGTM! Thanks very much! Please add a wiki URL like https://github.com/PaddlePaddle/Paddle/wiki/Operator-Development-Manual-Index (both Chinese and English) in next PR to show the details.

File: tools/check_api_approvals.sh [hunk 1]
```diff
@@ -68,13 +68,18 @@ if [ "$api_doc_spec_diff" != "" ]; then
...
+    check_approval 1 9301846 33742067 7913861
+fi
 
 for API_FILE in ${API_FILES[*]}; do
   API_CHANGE=`git diff --name-only upstream/$BRANCH | grep "${API_FILE}" | grep -v "/CMakeLists.txt" || true`
   echo "checking ${API_FILE} change, PR: ${GIT_PR_ID}, changes: ${API_CHANGE}"
   if [ "${API_CHANGE}" ] && [ "${GIT_PR_ID}" != "" ]; then
-      # NOTE: per_page=10000 should be ok for all cases, a PR review > 10000 is not human readable.
-      # approval_user_list: XiaoguangHu01 46782768,Xreki 12538138,luotao1 6836917,sneaxiy 32832641,qingqing01 7845005,guoshengCS 14105589,heavengate 12605721,kuke 3064195,Superjomn 328693,lanxianghit 47554610,cyj1986 39645414,hutuxian 11195205,frankwhzhang 20274488,nepeplwu 45024560,Dianhai 38231817,JiabinYang 22361972,chenwhql 22561442,zhiqiu 6888866,seiriosPlus 5442383,gongweibao 10721757,saxon-zh 2870059,Boyan-Liu 2870059. 
+      # NOTE: per_page=10000 should be ok for all cases, a PR review > 10000 is not human readable.q
```
Comment (luotao1) [inline tools/check_api_approvals.sh:81]: Why there is a `q` here? Maybe you can fix it in next PR.

Comment (Aurelius84) [reply tools/check_api_approvals.sh:81]: Er..., I will remove it in PR with wiki URL

**Comment by Aurelius84** [comment]
> LGTM! Thanks very much! Please add a wiki URL like https://github.com/PaddlePaddle/Paddle/wiki/Operator-Development-Manual-Index (both Chinese and English) in next PR to show the details.

OK, related wiki will be added in next PR.

**Pull request merged**